### ISSUE TITLE
Hyperparameter settings for single-gpu (or 8-gpus) training

### ISSUE THREAD

**Issue by mrsempress** [facebookresearch/univlg#6]
1. The paper mentions that the project used 32 A100 80G GPUs, with an effective batch size of 64. But I only have **one machine with 1 A100** and **another with 8 A100 80G GPUs**. How can I set the **batch size** and **max_iter** to achieve similar performance?
2. Also, the batch size on my 80G A100 only goes up to 8. What could be the reason for this?
3. If I have bs=8, num_gpus=8, then max_iter is $20000\times4\times8$? What is the **original training time**? Because the estimated training time on my machine is currently very long.

**Comment by ayushjain1144** [comment]
1. Hi, I do not have a definite answer for this, you may need to try a few things -- however, one thing you can try is to use gradient accumulation to mimic our effective batch size
2. `Also, the batch size on my 80G A100 only goes up to 8. What could be the reason for this?` You mean effective batch size of 8 on 8 GPUs? (i.e. bs=1 per GPU?). That is very strange. I think you should be able to feed at least 2 and possibly more (I think even 4 per GPU might work)
3. Our original training time is around 3-4 days. 

Other things to try, which we didn't get time to -- sample SR3D, NR3D, and ScanRefer in different ratios. Try sampling NR3D and ScanRefer more than SR3D (SR3D is a simpler dataset, and the model fits on it much faster). 

If you can, let us know here how things go for you, this might help future users with similar constraints.

**Comment by mrsempress** [comment]
1. You misunderstood me a little bit. "The batch size on my 80G A100 can only be up to 8" refers to **the batch size of 8 per A100**, just like the **BS** in your launcher refers to the batch size of each GPU. So, your original text "32 A100 80G GPUs, effective batch size of 64" also means the total batch size is 64, the batch size of each GPU is 2, and a total of 32 GPUs are used, right?
2. In case 1, when I use 8 A100 80G GPUs, my single batch size is set to 8, and the total batch size is 64. Don't I need to adjust the gradient accumulation? And the gradient accumulation cannot be set in main.sh, and I can't find the relevant settings in the project.
3. Under the setting of 2, I performed 3D-only training, using only the ScanRefer dataset, and the results are as follows:

| | Acc@25 | Acc@50 | Acc@75|
|---|---|---|---|
|paper | 60.7 | 53.2 | 42.6|
|Re-training | 55.9 | 47.7 | 36.7|

Or is it because Table 1 is trained on SR3D, NR3D, and ScanRefer datasets simultaneously and evaluated separately? If so, I would like to know what the training and test sets are for the other tables so that I can follow up on your work.

**Comment by ayushjain1144** [comment]
1. Correct. So, with 8 GPUs and with bs=8 per GPU, you should be using the same effective batch size as us, and you don't need to tune hyperparameters
2. You don't need gradient accumulation; we didn't experiment with gradient accumulation thoroughly. It exists in our code [though](https://github.com/facebookresearch/univlg/blob/main/train.py#L756C21-L768) but you don't need it. 
3. Yes, we train on SR3D, NR3D, ScanRefer,  ScanNet and Matterport3D for Table-1. See [here ](https://github.com/facebookresearch/univlg/blob/main/scripts/main.sh#L219). (I think if you skip scannet and matterport3d, it wouldn't affect the performance much -- just in case you don't have them setup).

**Comment by mrsempress** [comment]
Thank you for your reply. I will download Matterport3D and retrain to align the results. Thank you again for your patient reply.
